docs/sources/configure-client/trace-span-profiles/java-span-profiles.md

```diff
@@ -40,49 +40,52 @@ Your applications must be instrumented for profiling and tracing before you can
-    -javaagent:./opentelemetry-javaagent.jar \
-    -Dotel.javaagent.extensions=./pyroscope-otel.jar \
-    -Dotel.pyroscope.start.profiling=true \
-    -Dpyroscope.application.name=ride-sharing-app-java-instrumentation  \
-    -Dpyroscope.format=jfr \
-    -Dpyroscope.profiler.event=itimer \
-    -Dpyroscope.server.address=$PYROSCOPE_SERVER_ADDRESS \
-    # rest of your otel-java-instrumentation configuration
+```Dockerfile
+# [...]
 
+EXPOSE 5000
+
+CMD ["java", "-Dserver.port=5000", "-jar", "./my-app.jar" ]
 ```
 
-Next, you need to create and configure the tracer provider:
-```java
-implementation("io.pyroscope:otel:0.10.1.11")
+Now by adding the OTEL Java agent and our Pyroscope Otel Java Agent extensions, you can enrich your profiles with span IDs which makes it possible to query for span-specific profiling data in Grafana Tempo UI:
 
-// obtain SdkTracerProviderBuilder
-SdkTracerProviderBuilder tpBuilder = ...
+```Dockerfile
+# [...]
 
-// Add PyroscopeOtelSpanProcessor to SdkTracerProviderBuilder
-PyroscopeOtelConfiguration pyroscopeTelemetryConfig = new PyroscopeOtelConfiguration.Builder()
-  .setAddSpanName(true)
-  .setRootSpanOnly(true)
-  .build();
-tpBuilder.addSpanProcessor(new PyroscopeOtelSpanProcessor(pyroscopeOtelConfig));
+EXPOSE 5000
 
-```
+## Add required libararies
+ADD https://github.com/grafana/pyroscope-java/releases/download/v0.12.2/pyroscope.jar ./pyroscope.jar
+ADD https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v1.17.0/opentelemetry-javaagent.jar opentelemetry-javaagent.jar
+ADD https://repo1.maven.org/maven2/io/pyroscope/otel/0.10.1.11/otel-0.10.1.11.jar pyroscope-otel.jar
+
+ENV PYROSCOPE_APPLICATION_NAME=my-app
+ENV PYROSCOPE_FORMAT=jfr
+ENV PYROSCOPE_PROFILING_INTERVAL=10ms
+ENV PYROSCOPE_PROFILER_EVENT=itimer
+ENV PYROSCOPE_PROFILER_LOCK=10ms
+ENV PYROSCOPE_PROFILER_ALLOC=512k
+ENV PYROSCOPE_UPLOAD_INTERVAL=15s
+ENV OTEL_JAVAAGENT_EXTENSIONS=./pyroscope-otel.jar
+ENV OTEL_PYROSCOPE_ADD_PROFILE_URL=false
+ENV OTEL_PYROSCOPE_ADD_PROFILE_BASELINE_URL=false
+ENV OTEL_PYROSCOPE_START_PROFILING=true
+
+## Useful for debugging
+# ENV PYROSCOPE_LOG_LEVEL=debug
 
-Now that we set up the tracer, we can create a new trace from anywhere and the profiler will automatically capture profiles for it.
-```java
-Span span = tracer.spanBuilder("findNearestVehicle").startSpan();
-try (Scope s = span.makeCurrent()){
-    // Your code goes here.
-} finally {
-    span.end();
-}
+## Those environment variables need to be overwritten at runtime, if you are using Grafana Cloud 
+ENV PYROSCOPE_SERVER_ADDRESS=http://localhost:4040
+# ENV PYROSCOPE_BASIC_AUTH_USER=123     ## Grafana Cloud Username
+# ENV PYROSCOPE_BASIC_AUTH_PASSWORD=glc_secret ## Grafana Cloud Password / API Token
 
-// Your code goes here.
+## Add the pyroscope and the opentelemetry java-agents
+CMD ["java", "-Dserver.port=5000", "-javaagent:./opentelemetry-javaagent.jar", "-javaagent:pyroscope.jar", "-jar", "./my-app.jar" ]
```

Comment:
I can remember there was a discussion on whether we need to `-javaagent:pyroscope.jar` in the case of OTEL extension, or if it is optional. @korniltsev could you please clarify this?

Reply:
Yes I remember the discussion but not the outcome. Also @aleks-p might know. I will also give it an e2e try.

Reply:
The removal of the second agent was attempted with #3169, but in my tests this was fine at startup but as soon as we invoke an endpoint we got an error. Could be worth re-testing anyway.

Reply:
Yes verified, removing it causes this:

```
2024-06-21 13:42:18.421 ERROR 1 --- [nio-5000-exec-7] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Handler dispatch failed; nested exception is java.lang.UnsatisfiedLinkError: no asyncProfiler in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib] with root cause

java.lang.UnsatisfiedLinkError: no asyncProfiler in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2429) ~[na:na]
        at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818) ~[na:na]
        at java.base/java.lang.System.loadLibrary(System.java:1989) ~[na:na]
        at io.pyroscope.one.profiler.AsyncProfiler.getInstance(AsyncProfiler.java:50) ~[agent-0.13.1.jar!/:na]
        at io.pyroscope.one.profiler.AsyncProfiler.getInstance(AsyncProfiler.java:26) ~[agent-0.13.1.jar!/:na]
[...]

```